src/main/java/ml/shifu/shifu/core/alg/TensorflowTrainer.java

```diff
@@ -0,0 +1,144 @@
+import ml.shifu.shifu.container.obj.ModelConfig;
+import ml.shifu.shifu.container.obj.ModelTrainConf;
+import ml.shifu.shifu.core.dtrain.CommonConstants;
+import ml.shifu.shifu.fs.PathFinder;
+
+public class TensorflowTrainer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TensorflowTrainer.class);
+
+    private String inputDataPath;
+
+    private String outputModelDir;
+
+    private double learningRate = 1;
+
+    private List<Integer> seletectedColumnNums = new ArrayList<Integer>();
+
+    private int targetColumnNum = 0;
+
+    private List<String> actFuncs = new ArrayList<String>();
+
+    private List<Integer> hiddenLayerNodes = new ArrayList<Integer>();
+
+    private int hiddenLayers = 1;
+
+    private List<ColumnConfig> ccList;
+
+    private ModelConfig modelConfig;
+
+    private PathFinder pathFinder;
+
+    private String alg;
+
+    private char delimiter;
+
+    private String lossFunc;
+
+    private String optimizer;
+    
+    private int epoch = 100;
+
+    public TensorflowTrainer(ModelConfig modelConfig, List<ColumnConfig> ccList) {
+        this.modelConfig = modelConfig;
+        this.ccList = ccList;
+        this.pathFinder = new PathFinder(modelConfig);
+        
+        for(int i = 0; i < ccList.size(); i++) {
+            ColumnConfig cc = ccList.get(i);
+            if(cc.isTarget()) {
+                targetColumnNum = i;
+            } else if(cc.isFinalSelect()) {
+                seletectedColumnNums.add(i);
+            }
+        }
+        if(seletectedColumnNums.size() == 0) {
+            for(int i = 0; i < ccList.size(); i++) {
+                ColumnConfig cc = ccList.get(i);
+                if(cc.isTarget()) {
+                    continue;
+                }
+                seletectedColumnNums.add(i);
+            }
+        }
+
+        ModelTrainConf modelTrainConf = modelConfig.getTrain();
+        learningRate = (double) modelTrainConf.getParams().get(CommonConstants.LEARNING_RATE);
+        actFuncs = (List<String>) modelTrainConf.getParams().get(CommonConstants.ACTIVATION_FUNC);
+        hiddenLayerNodes = (List<Integer>) modelTrainConf.getParams().get(CommonConstants.NUM_HIDDEN_NODES);
+        hiddenLayers = hiddenLayerNodes.size();
+        inputDataPath = pathFinder.getNormalizedDataPath();
+        alg = modelConfig.getAlgorithm();
+        delimiter = modelConfig.getDataSetDelimiter().charAt(0);
+        lossFunc = (String) modelTrainConf.getParams().get(CommonConstants.TF_LOSS);
+        optimizer = (String) modelTrainConf.getParams().get(CommonConstants.TF_OPTIMIZER);
+        epoch = modelTrainConf.getNumTrainEpochs();
+    }
+
+    public void train() throws IOException {
+        List<String> commands = buildCommands();
+        ProcessBuilder pb = new ProcessBuilder(commands);
+        pb.directory(new File("./"));
+        LOGGER.info("Start trainning sub process. Commands {}", commands.toString());
+        Process p = pb.start();
```

Comment:
where could python console log re-directed? It is better to redirect console log to Java console and shifu.log.

Reply:
OK, I'll add a stream collector.